pekko-stream/src/main/scala/eu/neverblink/jelly/pekko/stream/impl/JellyIoOps.scala

```diff
@@ -48,6 +52,50 @@ object JellyIoOps:
     final def fromBytes: Flow[Array[Byte], RdfStreamFrame, NotUsed] =
       Flow[Array[Byte]].map(RdfStreamFrame.parseFrom)
 
+    /**
+     * Convert a stream of DELIMITED bytes into a stream of Jelly frames. The ByteStrings may be chunked
+     * in an arbitrary way, the stream will be framed based on the Protobuf varint-encoded length prefix.
+     *
+     * For example, one input ByteString may contain multiple Jelly frames, a part of a Jelly frame, or
+     * any combination of these.
+     *
+     * Using this method, you can read Jelly files from a file or socket in a fully reactive manner,
+     * without any blocking operations.
+     *
+     * @return Pekko Flow
+     */
+    final def fromByteStreamDelimited: Flow[ByteString, RdfStreamFrame, NotUsed] =
+      Flow[ByteString]
+        .via(protobufFraming(Int.MaxValue))
+        .map(byteString => {
+          val byteStrings = byteString.asByteBuffers
+          val codedInputStream = if byteStrings.size > 1 then
+            CodedInputStream.newInstance(byteStrings.map(_.asReadOnlyBuffer()).asJava)
+          else
+            CodedInputStream.newInstance(byteStrings.head.asReadOnlyBuffer())
+          val frame = RdfStreamFrame.newInstance()
+          ProtoMessage.mergeFrom(frame, codedInputStream, ProtoMessage.DEFAULT_MAX_RECURSION_DEPTH)
+        })
+
+    /**
+     * Frame Protobuf messages based on their varint-encoded length prefix.
+     *
+     * This flow reads the varint-encoded length prefix from the incoming ByteString and emits complete
+     * Protobuf messages as ByteStrings.
+     *
+     * The implementation is non-blocking and fully reactive.
+     *
+     * The method is made package-private only for use in tests.
+     *
+     * @param maxMessageSize Maximum allowed size for a Protobuf message.
+     *                       If a message exceeds this size, the stage will fail.
+     * @return
+     */
+    private[stream] final def protobufFraming(maxMessageSize: Int): Flow[ByteString, ByteString, NotUsed] =
```

Comment:
Consider making it public – would allow people to split the processing across an async boundary, for example. There is nothing "unsafe" about this method.